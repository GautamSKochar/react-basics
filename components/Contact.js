import React from 'react';
import ContactChild from './ContactChild';

class Contact extends React.Component {
    constructor() {
        super();
        this.state = {
            pageMounted: true
        };
    }

    onChangePageMounted() {
        this.setState({
            pageMounted: !this.state.pageMounted
        });
    };
    render() {
        let pageCmp = "";
        if (this.state.pageMounted) {
            pageCmp = <div>
                <ContactChild/>
            </div>;
        }
        return (
            <div>
                <h3>Contact Us</h3><br/>
                {pageCmp}<br/>
                <div>
                    <button onClick={this.onChangePageMounted.bind(this)} className="btn btn-primary">(Un)Mount Child Component</button>
                </div>
            </div>
        )
    }
}

export default Contact;