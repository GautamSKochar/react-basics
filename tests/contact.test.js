import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Contact from '../components/Contact';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}
configure({ adapter: new Adapter() });

test('Contact Component Should render as expected', () => {
    const component = shallow(<Contact />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});