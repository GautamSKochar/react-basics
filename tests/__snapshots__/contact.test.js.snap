// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component Should render as expected 1`] = `
<div>
  <h3>
    Contact Us
  </h3>
  <br />
  <div>
    <ContactChild />
  </div>
  <br />
  <div>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      (Un)Mount Child Component
    </button>
  </div>
</div>
`;
